### PR TITLE
Refactor published nodes configuration and persistency, changes and fixes in server methods

### PR DESCRIPTION
**Pull request by hansgschossmann**
### Refactor published node configuration handling
Put all node configuration code in separate module
Keep original configuration format
Configuration file updates produce better readable file format
### Refactor published node persistency
Publishers OPC data structures are the single point of truth
### Changes and fixes in publishers OPC UA server methods
Methods called with NodeId format node information will also target nodes configured with ExpandedNodeId format
Publishing requests with NodeId format will be persistent in ExpandedNodeId format

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -303,18 +303,26 @@ The Publisher OPC UA server listens by default on port 62222. To expose this inb
...
+To enable name resolution from within the container to other containers, you need to create a user define docker bridge network and connect the container to this network using the `--network` option.
 Additionally you need to assign the container a name using the `--name` option as in this example:
 
     docker network create -d bridge iot_edge
     docker run --network iot_edge --name publisher microsoft/iot-edge-opc-publisher <applicationname> [<iothubconnectionstring>] [options]
 
 The container can now be reached by other containers via the name `publisher`over the network.
 
+### Access other hosts from within the container
+Hosts which are implemented in other containers, could be reached with the parameters described in the "Enable intercontainer nameresolution" paragraph.
```
Comment (barnstee) [inline README.md]: can #Resolved

File: README.md [hunk 2]
```diff
@@ -303,18 +303,26 @@ The Publisher OPC UA server listens by default on port 62222. To expose this inb
...
+If the environment has DNS working, then accessing the host on which the docker runtime executes as was as any other DNS reachable host is no issue.
```
Comment (barnstee) [inline README.md]: well #Resolved

Comment (barnstee) [inline README.md]: is working #Resolved

Comment (mregen) [inline README.md]: > then accessing the host on which the docker runtime executes as was as any other DNS reachable host is no issue. [](start = 35, length = 113)

Confusing sentence :-) #Resolved

File: src/OpcSession.cs [hunk 3]
```diff
@@ -495,6 +585,117 @@ public OpcSession(Uri endpointUri, uint sessionTimeout)
             {
                 _opcSessionSemaphore.Release();
             }
+            return requestConfigFileUpdate;
+        }
+
+        /// <summary>
+        /// Checks if there are monitored nodes tagged to stop monitoring and stop monitoring.
```
Comment (mregen) [inline src/OpcSession.cs]: >to stop monitoring and stop monitoring. [](start = 55, length = 39)

stopstop? #Resolved

**Review by mregen — APPROVED** [review]
:shipit:

File: src/PublisherNodeManager.cs [hunk 4]
```diff
@@ -412,25 +430,18 @@ private MethodState CreateMethod(NodeState parent, uint id, string name)
         /// </summary>
         private ServiceResult OnPublishNodeCall(ISystemContext context, MethodState method, IList<object> inputArguments, IList<object> outputArguments)
```
Comment (mregen) [inline src/PublisherNodeManager.cs:431]: >     private ServiceResult OnPublishNodeCall [](start = 3, length = 44)

sometime we should investigate ho wto use async here... #ByDesign

**Review by barnstee — APPROVED** [review]
Few small typos.

File: src/PublisherNodeManager.cs (hunk 4 above)
Comment (hansgschossmann) [reply src/PublisherNodeManager.cs:431]: that would be great.

---
In reply to: [146244811](https://github.com/Azure/iot-edge-opc-publisher/pull/84#discussion_r146244811) [](ancestors = 146244811)

**Pull request merged**